src/utilities.cpp

```diff
@@ -559,6 +562,347 @@ void findAllVariablesWithEquivalences(const ComponentPtr &component, VariablePtr
     }
 }
 
+NameList findCnUnitsNames(const XmlNodePtr &node);
+NameList findComponentCnUnitsNames(const ComponentPtr &component);
+void findAndReplaceCnUnitsNames(const XmlNodePtr &node, const StringStringMap &replaceMap);
+void findAndReplaceComponentCnUnitsNames(const ComponentPtr &component, const StringStringMap &replaceMap);
+size_t getComponentIndexInComponentEntity(const ComponentEntityPtr &componentParent, const ComponentEntityPtr &component);
+IndexStack reverseEngineerIndexStack(const VariablePtr &variable);
+VariablePtr getVariableLocatedAt(const IndexStack &stack, const ModelPtr &model);
+void makeEquivalence(const IndexStack &stack1, const IndexStack &stack2, const ModelPtr &model);
+IndexStack rebaseIndexStack(const IndexStack &stack, const IndexStack &originStack, const IndexStack &destinationStack);
+void componentNames(const ComponentPtr &component, NameList &names);
+std::vector<UnitsPtr> referencedUnits(const ModelPtr &model, const UnitsPtr &units);
+
+NameList findCnUnitsNames(const XmlNodePtr &node)
+{
+    NameList names;
+    XmlNodePtr childNode = node->firstChild();
```

Comment:
Replace `XmlNodePtr` with `auto`. In fact, maybe we should include something about it in issue #402...?

Reply:
Yeah, you're only seeing these because it's appearing as "new" code, when really it's just been moved.  I'd rather address it en masse since it's not really related to the moving of the code ...

Reply:
Ok, fair enough.